agent/agentproc/proc.go

```diff
@@ -0,0 +1,127 @@
+package agentproc
+
+import (
+	"errors"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/spf13/afero"
+	"golang.org/x/xerrors"
+)
+
+const DefaultProcDir = "/proc"
```

Comment:
Since this is the only value ever passed into `List`, I think this constant should be `ProcDir` or even unexported and implicit.

This package is very linuxy and all linuxes use /proc.